services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightProducer.java

```diff
@@ -39,37 +52,54 @@
 import com.dremio.service.flight.impl.FlightPreparedStatement;
 import com.dremio.service.flight.impl.FlightWorkManager;
 import com.dremio.service.flight.impl.FlightWorkManager.RunQueryResponseHandlerFactory;
+import com.google.protobuf.Any;
 import com.google.protobuf.InvalidProtocolBufferException;
 
 /**
  * A FlightProducer implementation which exposes Dremio's catalog and produces results from SQL queries.
  */
 public class DremioFlightProducer implements FlightProducer {
+
+  private final DremioFlightSqlProducer flightSqlProducer;
+
   private final FlightWorkManager flightWorkManager;
   private final Location location;
   private final DremioFlightSessionsManager sessionsManager;
   private final BufferAllocator allocator;
 
   public DremioFlightProducer(Location location, DremioFlightSessionsManager sessionsManager,
                               Provider<UserWorker> workerProvider, Provider<OptionManager> optionManagerProvider,
-                              BufferAllocator allocator, RunQueryResponseHandlerFactory runQueryResponseHandlerFactory) {
+                              BufferAllocator allocator,
+                              RunQueryResponseHandlerFactory runQueryResponseHandlerFactory) {
     this.location = location;
     this.sessionsManager = sessionsManager;
     this.allocator = allocator;
 
     flightWorkManager = new FlightWorkManager(workerProvider, optionManagerProvider, runQueryResponseHandlerFactory);
+
+    flightSqlProducer =
+      new DremioFlightSqlProducer(location, sessionsManager, workerProvider, optionManagerProvider, allocator,
+        runQueryResponseHandlerFactory);
   }
 
   @Override
   public void getStream(CallContext callContext, Ticket ticket, ServerStreamListener serverStreamListener) {
+    if (isFlightSqlCommand(ticket)) {
```

Comment:
Would it make more sense to have this on `flightSqlProducer` as a static that should be removed once legacy is removed?